File: app/lib/format.dart
/// Small display helpers. Deliberately not a package: "3d" and a capitalised
/// provider name is the whole requirement.
library;

/// shortAgo renders a compact age: 5m, 3h, 12d. Anything older than a year is
/// rare enough on a job board to fall back to years.
String shortAgo(DateTime? time) {
  if (time == null) return '';
  final elapsed = DateTime.now().difference(time);
  if (elapsed.inMinutes < 1) return 'now';
  if (elapsed.inMinutes < 60) return '${elapsed.inMinutes}m';
  if (elapsed.inHours < 24) return '${elapsed.inHours}h';
  if (elapsed.inDays < 365) return '${elapsed.inDays}d';
  return '${elapsed.inDays ~/ 365}y';
}

/// providerLabel turns a registry key into something worth showing.
String providerLabel(String provider) => switch (provider) {
  'greenhouse' => 'Greenhouse',
  'lever' => 'Lever',
  'ashby' => 'Ashby',
  'smartrecruiters' => 'SmartRecruiters',
  'workable' => 'Workable',
  'recruitee' => 'Recruitee',
  _ => provider,
};

File: web/src/format.ts
/** Compact age: now, 5m, 3h, 12d, 2y. */
export function shortAgo(iso: string | null): string {
  if (!iso) return ''
  const ms = Date.now() - new Date(iso).getTime()
  const minutes = Math.floor(ms / 60_000)
  if (minutes < 1) return 'now'
  if (minutes < 60) return `${minutes}m`
  const hours = Math.floor(minutes / 60)
  if (hours < 24) return `${hours}h`
  const days = Math.floor(hours / 24)
  if (days < 365) return `${days}d`
  return `${Math.floor(days / 365)}y`
}

const PROVIDERS: Record<string, string> = {
  greenhouse: 'Greenhouse',
  lever: 'Lever',
  ashby: 'Ashby',
  smartrecruiters: 'SmartRecruiters',
  workable: 'Workable',
  recruitee: 'Recruitee',
  teamtailor: 'Teamtailor',
  phenom: 'Phenom',
  oracle: 'Oracle',
  himalayas: 'Himalayas',
  jobicy: 'Jobicy',
  jobven: 'Jobven',
  jobspipe: 'JobsPipe',
  workday: 'Workday',
}

export const providerLabel = (provider: string) => PROVIDERS[provider] ?? provider
